### PR TITLE
LIMS-3422 : Orders with test units: Edit the article and check that test unit is not affected 

### PR DESCRIPTION
**Pull request by modeso-NahlaMakhlouf**
Description : check that when edit order and change article , selected test unit shouldn't be affected 

Local Execution Result:
```
C:\Users\ADMIN\PycharmProjects\1lims-automation>nosetests -vs --nologcapture --tc-file=config.ini ui_testing/testcases/basic_tests/test006_orders.py:OrdersTestCases.test100_check_testunit
_after_change_article
C:\Users\ADMIN\AppData\Local\Programs\Python\Python38\lib\site-packages\nose\plugins\manager.py:394: RuntimeWarning: Unable to load plugin noseprogressive = noseprogressive:ProgressivePlu
gin: No module named '_curses'
  warn("Unable to load plugin %s: %s" % (ep, e),
2020-09-22 17:31:50.406 | INFO     | api_testing.apis.base_api:_get_authorized_session:46 - Get authorized api session.
2020-09-22 17:31:50.408 | INFO     | api_testing.apis.base_api:_get_authorized_session:47 - admin:admin
2020-09-22 17:31:51.086 | INFO     | api_testing.apis.base_api:_get_authorized_session:58 - session ID : j2z_82YDELTurSaCWECK6K2Gej1sP7fEMjvp_ck08lo .....

DevTools listening on ws://127.0.0.1:53162/devtools/browser/626c7955-3abd-4df6-b66b-65c090f1ea8d
Orders with test units: Edit the article and check that test unit is not affected ...
2020-09-22 17:32:24.198 | INFO     | ui_testing.testcases.base_test:setUp:27 - Test case : test100_check_testunit_after_change_article
2020-09-22 17:32:26.933 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:603 - wait until page is loaded
2020-09-22 17:32:26.962 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 17:32:27.707 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 17:32:28.432 | INFO     | api_testing.apis.test_unit_api:set_name_configuration_name_only:437 - set test unit configuration
2020-09-22 17:32:29.036 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-22 17:32:29.044 | INFO     | api_testing.apis.orders_api:set_configuration:498 - set order configuration
2020-09-22 17:32:29.585 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-22 17:32:29.590 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test100_check_testunit_after_change_article:3491 - create new order
2020-09-22 17:32:29.592 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits
2020-09-22 17:32:30.075 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 17:32:30.085 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits/get/242
2020-09-22 17:32:30.201 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 17:32:30.211 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/orders/auto/generatedOrderNumber/?yearOption=1
2020-09-22 17:32:30.326 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 17:32:30.329 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/articles
2020-09-22 17:32:30.599 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 17:32:30.602 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/materialTypes
2020-09-22 17:32:30.708 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 17:32:30.717 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits
2020-09-22 17:32:30.870 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 17:32:30.873 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits/get/243
2020-09-22 17:32:31.000 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 17:32:31.009 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testPlans
2020-09-22 17:32:31.156 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 17:32:31.165 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits
2020-09-22 17:32:31.320 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 17:32:31.330 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits/get/244
2020-09-22 17:32:31.440 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 17:32:31.442 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/contacts
2020-09-22 17:32:31.560 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 17:32:31.575 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/orders
2020-09-22 17:32:31.832 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 17:32:31.842 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test100_check_testunit_after_change_article:3494 - get order test unit
2020-09-22 17:32:31.863 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test100_check_testunit_after_change_article:3496 - edit order and update article
2020-09-22 17:32:33.074 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:603 - wait until page is loaded
2020-09-22 17:32:33.570 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 17:32:34.339 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 17:32:35.015 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/articles
2020-09-22 17:32:35.326 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 17:32:35.768 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-22 17:32:37.090 | INFO     | ui_testing.pages.order_page:update_suborder:525 - Set article name : 24becbe127
2020-09-22 17:32:46.364 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-22 17:32:48.674 | INFO     | ui_testing.pages.order_page:update_suborder:531 -  Set test plan : [] for 0 time(s)
2020-09-22 17:32:48.677 | INFO     | ui_testing.pages.order_page:update_suborder:540 -  Set test unit : [] for 0 time(s)
2020-09-22 17:32:48.678 | INFO     | ui_testing.pages.base_pages:save:55 - save the changes
2020-09-22 17:32:48.679 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 17:32:49.452 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 17:32:50.158 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test100_check_testunit_after_change_article:3502 - get test unit after update article
2020-09-22 17:32:50.198 | INFO     | ui_testing.pages.order_page:get_suborder_data:432 - getting main order data
2020-09-22 17:32:50.268 | INFO     | ui_testing.pages.order_page:get_suborder_data:439 - getting suborders data
2020-09-22 17:32:50.516 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test100_check_testunit_after_change_article:3505 - Assert that test unit doesnt affected by updating a
rticle
2020-09-22 17:32:50.518 | INFO     | ui_testing.testcases.base_test:tearDown:33 - go to dashboard page
2020-09-22 17:32:51.582 | INFO     | ui_testing.testcases.base_test:tearDown:35 - TearDown.     
ok

----------------------------------------------------------------------
Ran 1 test in 62.261s

OK

```

### CODE REVIEW

File: api_testing/apis/orders_api.py [hunk 1]
```diff
@@ -512,7 +512,7 @@ def create_order_with_multiple_suborders_double_tp(self, no_suborders=3):
         suborders = []
         for _ in range(no_suborders):
             suborder = {}
-            created_suborder_data = TestPlanAPI().create_multiple_test_plan_with_same_article(no_of_testplans=2)
+            created_suborder_data = TestPlanAPI().create_multiple_test_plan_with_same_article()
```
Comment (0xIslamTaha) [inline api_testing/apis/orders_api.py:515]: what is the purpose of changing it?

File: api_testing/apis/test_plan_api.py [hunk 2]
```diff
@@ -393,19 +394,20 @@ def set_configuration(self):
             payload = json.load(read_file)
         super().set_configuration(payload=payload)
 
-    def create_multiple_test_plan_with_same_article(self, no_of_testplans=2):
+    def create_multiple_test_plan_with_same_article(self):
```
Comment (0xIslamTaha) [inline api_testing/apis/test_plan_api.py:397]: use the one in master.

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 3]
```diff
@@ -3538,6 +3539,29 @@ def test100_create_multiple_suborders_with_testplans_testunits(self):
...
+         LIMS-3422
+               """
+        self.info('create new order')
+        res, payload = self.orders_api.create_order_with_test_units(no_of_test_units=1)
+        self.assertEqual(res['status'], 1)
+        self.info('get order test unit')
+        selected_testunits = payload[0]['testUnits'][0]['name']
+        self.info('edit order and update article')
+        self.order_page.get_order_edit_page_by_id(id=res['order']['mainOrderId'])
+        api, _payload = self.article_api.get_all_articles()
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test006_orders.py:3555]: `res, _ = self.article_api.get_all_articles()`

**Review by 0xIslamTaha — CHANGES_REQUESTED** [review]
please, do the commentes

File: ui_testing/testcases/basic_tests/test006_orders.py (hunk 3 above)
Comment (modeso-NahlaMakhlouf) [reply ui_testing/testcases/basic_tests/test006_orders.py:3555]: done

**Pull request closed without merging**